bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,767 @@
+        )
+        registry_id = repository["repository"]["registryId"]
+
+    except ecr_client.exceptions.RepositoryAlreadyExistsException:
+        all_repositories = ecr_client.describe_repositories(repositoryNames=[repo_name])
+        registry_id = all_repositories["repositories"][0]["registryId"]
+
+    return registry_id
+
+
+def _get_ecr_password(registry_id, region):
+    ecr_client = boto3.client("ecr", region)
+    try:
+        token_data = ecr_client.get_authorization_token(registryIds=[registry_id])
+        token = token_data["authorizationData"][0]["authorizationToken"]
+        registry_endpoint = token_data["authorizationData"][0]["proxyEndpoint"]
+        return token, registry_endpoint
+
+    except ClientError as error:
+        if (
+            error.response
+            and error.response["Error"]["Code"] == "InvalidParameterException"
+        ):
+            raise BentoMLException(
+                "Could not get token for registry {registry_id},{error}".format(
+                    registry_id=registry_id, error=error.response["Error"]["Message"]
+                )
+            )
+
+
+def _get_creds_from_token(token):
+    cred_string = base64.b64decode(token).decode("ascii")
+    username, password = str(cred_string).split(":")
+    return username, password
+
+
+def _make_user_data(registry, tag, region):
+    base_format = """MIME-Version: 1.0
+Content-Type: multipart/mixed; boundary=\"==MYBOUNDARY==\"
+
+--==MYBOUNDARY==
+Content-Type: text/cloud-config; charset=\"us-ascii\"
+
+runcmd:
+
+- sudo yum update -y
+- sudo amazon-linux-extras install docker -y
+- sudo service docker start
+- sudo usermod -a -G docker ec2-user
+- curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+- unzip awscliv2.zip
+- sudo ./aws/install
+- ln -s /usr/bin/aws aws
+- aws ecr get-login-password --region {region} | docker login --username AWS --password-stdin {registry}
+- docker pull {tag}
+- docker run -p 5000:5000 {tag}
+
+--==MYBOUNDARY==--
+""".format(
+        registry=registry, tag=tag, region=region
+    )
+    encoded = base64.b64encode(base_format.encode("ascii")).decode("ascii")
+    return encoded
+
+
+def _make_cloudformation_template(
+    project_dir,
+    user_data,
+    s3_bucket_name,
+    sam_template_name,
+    ami_id,
+    instance_type,
+    autoscaling_min_size,
+    autoscaling_desired_size,
+    autoscaling_max_size,
+):
+    """
+    NOTE: SSH ACCESS TO INSTANCE MAY NOT BE REQUIRED
+    TODO: Port taken from cli
+    """
+    if (
+        autoscaling_min_size < 0
+        or autoscaling_max_size < autoscaling_min_size
+        or autoscaling_desired_size < autoscaling_min_size
+        or autoscaling_desired_size > autoscaling_max_size
+    ):
+        raise BentoMLException(
+            "Wrong autoscaling capacity specified. "
+            "It should be min_size <= desired_size <= max_size"
+        )
```

Comment:
nitpick, we can move this to `bentoml.yatai.validator.deployment_pb_validator`.

Reply:
fixed